sources/core.ts

```diff
@@ -458,9 +458,9 @@ export const reducers = {
 // ------------------------------------------------------------------------
 
 export type ArityDefinition = {
-    leading: number;
-    extra: number;
-    trailing: number;
+    leading: string[];
+    extra: string[] | undefined;
```

Comment:
Not a fan of overloading the type signature since it isn't actually "undefined". But if it works it works.